### ISSUE TITLE
SVG v3 highlighting is never removed

### ISSUE THREAD

**Issue by dpvc** [zorkow/speech-rule-engine#409]
In the V3 highlighter for SVG, the `rect` elements that are inserted for the background highlighting are never removed.  (See mathjax/MathJax#2529.)  This is due to the fact that the `setHighlighted()` method in the original SVG highlighter doesn't mark the highlighted node properly.  Since the V3 highlighter is a subclass of the original v2 one, it inherits the broken method.  The broken method seems to mark the node only when it is an SVG node.

**Comment by zorkow** [comment]
That is a regression that was introduced in May when I made the latest versions of the highlighters compatible with MJ2.
Previously we had one instance of a highlighter running in the page and there was a global state which node was highlighted. Now we have one instance per expression, having it's own state. In v3 we can have multiple highlighters running on the same expression, and we don't want them to highlight a node again (normally not a problem, but for SVG we generally introduce a new `rect` node for highlighting, and we do not want that to happen multiple times). 
In addition, in v3 we need to remember the `g` or `use` node instead of the `rect` as the latter could have already been removed by `unhighlight` of a different explorer, which might have run first.

Since introduced `rect` come with their own attributes, we only need to set the attribute explicitly for the top level `svg` node. So the v2 highlighter method is not broken, but it now interfered with the v3 method. 
So adding the attribute to the highlighted node for v3 does solve the problem. 

I might be able to simplify the v2 highlighter by using the same mechanism as for v3, but I'll leave this for a time when we create another v2.7 patch.
